mars/dataframe/datasource/read_sql.py

```diff
@@ -0,0 +1,647 @@
+        new_op = op.copy()
+        return new_op.new_dataframes(None, chunks=out_chunks, nsplits=nsplits,
+                                     **df.params)
+
+    def _parse_datetime(self, val):
+        if isinstance(self.parse_dates, list):
+            return pd.to_datetime(val)
+        args = self.parse_dates[self.partition_col]
+        args = {'format': args} if isinstance(args, str) else args
+        return pd.to_datetime(val, **args)
+
+    @classmethod
+    def _tile_partition(cls, op: 'DataFrameReadSQL'):
+        df = op.outputs[0]
+
+        selectable, _ = op._get_selectable(None)
+
+        if op._low_limit is None or op._high_limit is None:
+            import sqlalchemy as sa
+            from sqlalchemy import sql
+
+            engine = sa.create_engine(op.con, **(op.engine_kwargs or dict()))
+            try:
+                part_col = selectable.columns[op.partition_col]
+                range_results = engine.execute(sql.select([sql.func.min(part_col), sql.func.max(part_col)]))
+
+                op._low_limit, op._high_limit = next(range_results)
+                if op.parse_dates and op.partition_col in op.parse_dates:
+                    op._low_limit = op._parse_datetime(op._low_limit)
+                    op._high_limit = op._parse_datetime(op._high_limit)
+            finally:
+                engine.dispose()
+
+        if isinstance(op._low_limit, (datetime.datetime, np.datetime64, pd.Timestamp)):
+            start_val = pd.Timestamp(op._low_limit).value
+            end_val = pd.Timestamp(op._high_limit).value
+            seps = pd.to_datetime(np.linspace(start_val, end_val, op.num_partitions + 1, endpoint=True), utc=True)
+        else:
+            seps = np.linspace(op._low_limit, op._high_limit, op.num_partitions + 1, endpoint=True)
+
+        out_chunks = []
+        for i, (start, end) in enumerate(zip(seps, seps[1:])):
+            chunk_op = op.copy().reset_key()
+            chunk_op._row_memory_usage = None  # no need for chunk
+            chunk_op._num_partitions = None
+            chunk_op._low_limit = start
+            chunk_op._high_limit = end
+            chunk_op._left_end = i == 0
+            chunk_op._right_end = i == op.num_partitions - 1
+
+            if df.index_value.has_value():
+                # range index
+                index_value = parse_index(-1, chunk_op.key, chunk_op.index_value.key)
+            else:
+                index_value = parse_index(df.index_value.to_pandas(),
+                                          str(selectable), op.con, i)
+            out_chunk = chunk_op.new_chunk(None, shape=(np.nan, df.shape[1]),
+                                           columns_value=df.columns_value,
+                                           index_value=index_value, dtypes=df.dtypes,
+                                           index=(i, 0))
+            out_chunks.append(out_chunk)
+
+        out_chunks = standardize_range_index(out_chunks)
```

Comment:
We should add  an argument like `incremental_index`, and set false by default, because it's a heavy operation for `standardize_range_index`.